pkg/operator/controller/dns/controller.go

```diff
@@ -113,15 +113,26 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 			return reconcile.Result{RequeueAfter: 15 * time.Second}, nil
 		}
 	}
-
 	var zones []configv1.DNSZone
 	if dnsConfig.Spec.PrivateZone != nil {
 		zones = append(zones, *dnsConfig.Spec.PrivateZone)
 	}
 	if dnsConfig.Spec.PublicZone != nil {
 		zones = append(zones, *dnsConfig.Spec.PublicZone)
 	}
+	statuses, result := r.publishRecordToZones(zones, record)
+	// TODO: only update if status changed
+	updated := record.DeepCopy()
+	updated.Status.Zones = statuses
+	if err := r.client.Status().Update(context.TODO(), updated); err != nil {
+		log.Error(err, "failed to update dnsrecord; will retry", "dnsrecord", record)
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
```

Comment:
Just a suggestion: I'd omit the blank line so as to group `return result, nil` with the declaration of `result`.

Reply:
OK I will remove the blank line